File: ONCE_A_WEEK/WEEK6/곽태웅/20.js
function solution(participant, completion) {
    let map = new Map();
        pl = participant.length;
        result = '';
    
    for (let i = 0; i < pl; i++) {
        let p = participant[i];
            c = completion[i];
        
        map.set(p, (map.get(p) || 0) + 1);
        map.set(c, (map.get(c) || 0) - 1);
    }
    
    map.forEach((v, k) => {
        if (v > 0) result = k;
    })
    
    return result;
}

// 자바스크립트로는 코테보는 게 아니다...
// 과거에 풀었던 기록이 남아있어서 추가

File: ONCE_A_WEEK/WEEK6/최형욱/20.py
def solution(participant, completion):
    dic = {}

    for p in participant:
        if p in dic:
            dic[p] += 1
        else:
            dic[p] = 1

    for c in completion:
        dic[c] -= 1

    for key in dic.keys():
        if dic[key] > 0:
            return key
        
print(solution(["leo", "kiki", "eden"], ["eden", "kiki"]))
print(solution(["marina", "josipa", "nikola", "vinko", "filipa"], ["josipa", "filipa", "marina", "nikola"]))
print(solution(["mislav", "stanko", "mislav", "ana"], ["stanko", "ana", "mislav"]))